exchanges/coinbase/coinbase_test.go

```diff
@@ -0,0 +1,2054 @@
+	_, err = createOrderConfig(params)
+	assert.ErrorIs(t, err, errEndTimeInPast)
+	params.OrderType = order.StopLimit
+	_, err = createOrderConfig(params)
+	assert.NoError(t, err)
+	params.EndTime = time.Unix(1, 1)
+	_, err = createOrderConfig(params)
+	assert.ErrorIs(t, err, errEndTimeInPast)
+	params.EndTime = time.Now().Add(time.Hour)
+	_, err = createOrderConfig(params)
+	assert.NoError(t, err)
+	params.OrderType = order.Bracket
+	_, err = createOrderConfig(params)
+	assert.NoError(t, err)
+	params.EndTime = time.Unix(1, 1)
+	_, err = createOrderConfig(params)
+	assert.ErrorIs(t, err, errEndTimeInPast)
+	params.EndTime = time.Time{}
+	_, err = createOrderConfig(params)
+	assert.NoError(t, err)
+}
+
+func TestFormatMarginType(t *testing.T) {
+	t.Parallel()
+	resp := FormatMarginType("ISOLATED")
+	assert.Equal(t, "ISOLATED", resp)
+	resp = FormatMarginType("MULTI")
+	assert.Equal(t, "CROSS", resp)
+	resp = FormatMarginType("fake")
+	assert.Empty(t, resp)
+}
+
+func TestStatusToStandardStatus(t *testing.T) {
+	t.Parallel()
+	resp, _ := statusToStandardStatus("PENDING")
+	assert.Equal(t, order.New, resp)
+	resp, _ = statusToStandardStatus("OPEN")
+	assert.Equal(t, order.Active, resp)
+	resp, _ = statusToStandardStatus("FILLED")
+	assert.Equal(t, order.Filled, resp)
+	resp, _ = statusToStandardStatus("CANCELLED")
+	assert.Equal(t, order.Cancelled, resp)
+	resp, _ = statusToStandardStatus("EXPIRED")
+	assert.Equal(t, order.Expired, resp)
+	resp, _ = statusToStandardStatus("FAILED")
+	assert.Equal(t, order.Rejected, resp)
+	_, err := statusToStandardStatus("")
+	assert.ErrorIs(t, err, order.ErrUnsupportedStatusType)
+}
+
+func TestStringToStandardType(t *testing.T) {
+	t.Parallel()
+	resp, _ := stringToStandardType("LIMIT_ORDER_TYPE")
+	assert.Equal(t, order.Limit, resp)
+	resp, _ = stringToStandardType("MARKET_ORDER_TYPE")
+	assert.Equal(t, order.Market, resp)
+	resp, _ = stringToStandardType("STOP_LIMIT_ORDER_TYPE")
+	assert.Equal(t, order.StopLimit, resp)
+	_, err := stringToStandardType("")
+	assert.ErrorIs(t, err, order.ErrUnrecognisedOrderType)
+}
+
+func TestStringToStandardAsset(t *testing.T) {
+	t.Parallel()
+	resp, _ := stringToStandardAsset("SPOT")
+	assert.Equal(t, asset.Spot, resp)
+	resp, _ = stringToStandardAsset("FUTURE")
+	assert.Equal(t, asset.Futures, resp)
+	_, err := stringToStandardAsset("")
+	assert.ErrorIs(t, err, asset.ErrNotSupported)
+}
+
+func TestStrategyDecoder(t *testing.T) {
+	t.Parallel()
+	resp, _ := strategyDecoder("IMMEDIATE_OR_CANCEL")
+	assert.True(t, resp.Is(order.ImmediateOrCancel))
+	resp, _ = strategyDecoder("FILL_OR_KILL")
+	assert.True(t, resp.Is(order.FillOrKill))
+	resp, _ = strategyDecoder("GOOD_UNTIL_CANCELLED")
+	assert.True(t, resp.Is(order.GoodTillCancel))
+	resp, _ = strategyDecoder("GOOD_UNTIL_DATE_TIME")
+	assert.True(t, resp.Is(order.GoodTillDay|order.GoodTillTime))
+	_, err := strategyDecoder("")
+	assert.ErrorIs(t, err, errUnrecognisedStrategyType)
+}
+
+func TestBase64URLEncode(t *testing.T) {
```

Comment:
Can be removed now since you're using base64 rawdog encoding which does the same thing